tempesta_fw/http_msg.c

```diff
@@ -45,13 +45,28 @@ tfw_http_msg_hdr_val(TfwStr *hdr, int id, TfwStr *val)
 	TfwStr *c;
 	int nlen = hdr_lens[id];
 
-	BUG_ON(TFW_STR_PLAIN(hdr));
 	BUG_ON(TFW_STR_DUP(hdr));
-	BUG_ON(nlen >= hdr->len);
 	BUG_ON(id >= TFW_HTTP_HDR_RAW);
 
+	/* Only Host: header is allowed to be empty */
+	if (id == TFW_HTTP_HDR_HOST) {
```

Comment:
There are cases where `BUG_ON()` may be compiled out. That would leave an empty block here. Please put everything in `BUG_ON()` statements.